charts/spinnaker/templates/_helpers.tpl

```diff
@@ -63,9 +63,15 @@ Redis base URL for Spinnaker
 */}}
 {{- define "spinnaker.redisBaseURL" -}}
 {{- if .Values.redis.enabled }}
+{{- if .Values.redis.password }}
 {{- printf "redis://:%s@%s-redis-master:6379" .Values.redis.password .Release.Name -}}
+{{- else if .Values.redis.auth.existingSecret }}
```

Comment:
Should this be something like
```suggestion
{{- else if and not .Values.redis.external.host .Values.redis.auth.existingSecret }}
```

Otherwise, the case on line 71 is not reachable, right?

Reply:
This is a bit unintuitive but it is still reachable IIUC. The idea behind `redis.enabled` is that it deploys Redis via the (sub)chart -- if the flag is set to false, we assume that you have an external Redis (this is how the chart has originally been set up by OpsMx, so we follow that convention). In other words, `{{- else if .Values.redis.external.password }}` matches the `{{- if .Values.redis.enabled }}` clause (this is hard to read in a Helm template unfortunately). Does that make sense?

Reply:
Ah, damn, I missed that the second `{{- else if .Values.redis.auth.existingSecret }}` is on the different `if-else` level. 🤦‍♂️ This syntax without indentation is absolutely not readable...

Reply:
Totally agreed. When I wrote this I added indentation myself so I could see what's going on, but to match the style I had to revert that 🙈